packages/happo-target-react-native/src/native/StoryRenderer.js

```diff
@@ -0,0 +1,65 @@
+import React, { PropTypes } from 'react';
+import {
+  StyleSheet,
+  View,
+  Dimensions,
+} from 'react-native';
+import { takeSnapshot } from 'react-native-view-shot';
+
+const propTypes = {
+  name: PropTypes.string.isRequired,
+  Component: PropTypes.func.isRequired,
+  onSnapshotCompleted: PropTypes.func.isRequired,
+};
+
+const styles = StyleSheet.create({
+  container: {
+    backgroundColor: '#fff',
+  },
+});
+
+class StoryRenderer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.setRef = this.setRef.bind(this);
+    this.onLayout = this.onLayout.bind(this);
+    const { width } = Dimensions.get('window');
+    this.state = {
+      width,
+    };
+  }
+  onLayout() {
+    takeSnapshot(this.box, {
+      format: 'png',
+      quality: 1.0,
+      result: 'data-uri',
+    })
+      .then((uri) => {
+        this.props.onSnapshotCompleted(this.props.name, uri);
+      })
+      .catch((err) => {
+        console.error(err);
+        // TODO(lmr): this shouldn't happen, but we should expose something to user maybe?
```

Comment:
i guess the todo here is kind of redundant. `console.error` in RN will actually expose something to the user. Though maybe we could do more and send some info back to the server and console.log it back in the server process so it shows up in CI logs